src/gridfs.rs

```diff
@@ -318,28 +329,58 @@ impl GridFsBucket {
         todo!()
     }
 
-    /// Given an `id`, deletes the stored file's files collection document and
-    /// associated chunks from a [`GridFsBucket`].
-    pub async fn delete(&self, id: Bson) {
-        todo!()
+    /// Deletes the [`FilesCollectionDocument`] with the given `id `and its associated chunks from
+    /// this bucket.
+    pub async fn delete(&self, id: Bson) -> Result<()> {
+        let delete_result = self
+            .files()
+            .delete_one(doc! { "_id": id.clone() }, None)
```

Comment:
Just to validate - `delete` doesn't return an error if zero documents match, correct?

Reply:
yep, just verified in the shell